exchanges/binance/binance.go

```diff
@@ -161,27 +162,89 @@ func (b *Binance) GetHistoricalTrades(symbol string, limit int, fromID int64) ([
 	return nil, common.ErrFunctionNotSupported
 }
 
-// GetAggregatedTrades returns aggregated trade activity
-//
-// symbol: string of currency pair
-// limit: Optional. Default 500; max 1000.
-func (b *Binance) GetAggregatedTrades(symbol string, limit int) ([]AggregatedTrade, error) {
-	var resp []AggregatedTrade
-
-	if err := b.CheckLimit(limit); err != nil {
-		return resp, err
-	}
-
+// GetAggregatedTrades returns aggregated trade activity.
+// If more than one hour of data is requested with no limit, the trades are collected with multiple backend requests.
+// https://binance-docs.github.io/apidocs/spot/en/#compressed-aggregate-trades-list
+func (b *Binance) GetAggregatedTrades(arg *AggregatedTradeRequestParams) ([]AggregatedTrade, error) {
 	params := url.Values{}
-	params.Set("symbol", strings.ToUpper(symbol))
-	if limit > 0 {
-		params.Set("limit", strconv.Itoa(limit))
+	symbol := arg.Symbol
+	symbol.Delimiter = ""
+	params.Set("symbol", strings.ToUpper(symbol.String()))
+	if arg.Limit > 0 {
+		if err := b.CheckLimit(arg.Limit); err != nil {
+			return nil, err
+		}
+		params.Set("limit", strconv.Itoa(arg.Limit))
+	}
+	if arg.FromID != 0 {
+		params.Set("fromId", strconv.FormatInt(arg.FromID, 10))
+	}
+	if !arg.StartTime.IsZero() {
+		params.Set("startTime", strconv.FormatInt(convert.UnixMillis(arg.StartTime), 10))
+	}
+	if !arg.EndTime.IsZero() {
+		params.Set("endTime", strconv.FormatInt(convert.UnixMillis(arg.EndTime), 10))
+	}
+	// If startTime and endTime are sent, time between startTime and endTime must be less than 1 hour.
+	if arg.Limit == 0 && !arg.StartTime.IsZero() && !arg.EndTime.IsZero() && arg.EndTime.Sub(arg.StartTime) > time.Hour {
```

Comment:
theoretically if someone wanted to get a maximum of 500 trades OR all trades within past 2 hours, then this if condition becomes an issue since you are restricting people to take all the trades within a given timeframe or not setting a timeframe.

Reply:
Good catch, originally I was thinking keeping the limit to mean limit per request, but it makes sense to keep the semantics of "maximum number of trades to return" for the batch request